instaclustr/resource_cluster.go

```diff
@@ -438,6 +610,14 @@ func resourceClusterCreate(d *schema.ResourceData, meta interface{}) error {
 		createData.RackAllocation = &rackAllocation
 	}
 
+	// for multi-DC cluster
+	if len(dataCentres) > 1 {
+		createData.RackAllocation = createData.DataCentres[0].RackAllocation
```

Comment:
[question] can we rely on the ordering here? The `DataCentres` collection is `schema.TypeSet`, could we get ourselves into a strange situation here?

> TypeSet implements set behavior and is used to represent an unordered collection of items, meaning that their ordering specified does not need to be consistent, and the ordering itself has no impact on the behavior of the resource.
> 
> The elements of a set can be any of the other types allowed by Terraform, including another schema. Set items cannot be repeated.

https://www.terraform.io/docs/extend/schemas/schema-types.html#typeset